code-challenges.js

```diff
@@ -2,23 +2,109 @@
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
-var secretCodeWord1 = "lackadaisical"
+var secretCodeWord1 = 'lackadaisical'
 // Expected output: "l4ck4d41s1c4l"
-var secretCodeWord2 = "gobbledygook"
+var secretCodeWord2 = 'gobbledygook'
 // Expected output: "g0bbl3dyg00k"
 
-
-
-
+// FIRST CODE
+// Create a function that takes a string
+const secretCodeConverter = (str) => {
+	// convert string to an array - held in variable
+	// create an array to hold new values
+	// map over array
+	let newArray = str.split('').map((letter) => {
+		// check each letter and change based on letter being a, e, i or o
+		if (letter === 'a' || letter === 'A') {
+			return '4'
+		} else if (letter === 'e' || letter === 'E') {
+			return '3'
+		} else if (letter === 'i' || letter === 'I') {
+			return '1'
+		} else if (letter === 'o' || letter === 'O') {
+			return '0'
+		} else {
+			return letter
+		}
+	})
+	// convert array back to string (join(''))
+	// return a coded message with vowels converted to ('a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0)
+	return newArray.join('')
+}
+
+// REFACTOR ORIGINAL CODE
+const secretCodeConverter2 = (str) => {
+	// create an array to hold new values
+	// convert string to an array - held in variable
+	// map over array
+	let newArray = str.split('').map((letter) => {
+		// check each letter and change based on letter being a, e, i or o
+		if (letter === 'a' || letter === 'A') {
+			return '4'
+		} else if (letter === 'e' || letter === 'E') {
+			return '3'
+		} else if (letter === 'i' || letter === 'I') {
+			return '1'
+		} else if (letter === 'o' || letter === 'O') {
+			return '0'
+		} else {
+			return letter
+		}
+	})
+	// convert array back to string (join(''))
+	// return a coded message with vowels converted to ('a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0)
+	return newArray.join('')
+}
+
+// call the function
+console.log('Problem #1')
+console.log(secretCodeConverter(secretCodeWord1))
+console.log(secretCodeConverter(secretCodeWord2))
+console.log('Problem #1 - Refactor')
+console.log(secretCodeConverter2(secretCodeWord1))
+console.log(secretCodeConverter2(secretCodeWord2))
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
-var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
+var arrayOfWords = ['Apple', 'Banana', 'Plum', 'Cherry', 'Kiwi', 'Peach']
 // Expected output: "Apple" "Banana" "Peach"
 
-
-
-
+// Create a function that takes an array of words as an argument
+const wordsWithA = (arr) => {
+	// create a variable to hold new array of words with 'a'
+	let aWords = []
+
+	// iterate over each word in the array
+	arr.map((word) => {
+		// determine if word includes the letter a
+		if (word.includes('a') || word.includes('A')) {
```

Comment:
Stretch goal: for refactoring, is there a way to check for both the lowercase and uppercase of a letter without having to hard code both cases?